setupext.py

```diff
@@ -289,27 +290,12 @@ def __init__(self):
             except KeyError:
                 self.pkg_config = 'pkg-config'
 
-            self.set_pkgconfig_path()
             self.has_pkgconfig = shutil.which(self.pkg_config) is not None
             if not self.has_pkgconfig:
                 print("IMPORTANT WARNING:\n"
                       "    pkg-config is not installed.\n"
                       "    matplotlib may not be able to find some of its dependencies")
 
-    def set_pkgconfig_path(self):
```

Comment:
Why was this removed?
Note that (unchecked...) this may be needed to make things work in a conda env which does install the libraries in the environment but where pkg-config is "outside" the env (because sysconfig.get_config_var("LIBDIR") will correctly point inside the env in that case).

Reply:
i explained in the commit message.  did you see that ?

Reply:
OK, but in that case I would like someone (perhaps me, just not now) to check whether building Matplotlib as follows still works: in a conda env, freetype and libpng installed from conda, pkg-config not installed from conda but available outside of conda (e.g. from the distro).  Also, what gets linked (the conda libs or the distro libs) in the shared objects in that case?
I'm not saying this won't work or that this absolutely needs to work, but if it used to and doesn't anymore, then one should make sure that the docs clearly state that a conda pkg-config is needed in that setup.

Reply:
Personally, I usually don't read the commit messages.  We usually include important info in the PR body above and/or the code itself.  We usually try and keep the commit messages quite minimal.  Thanks!

Reply:
(I don't think that "trying to keep commit messages minimal" is a good advice (quite the contrary...).  But yes, it's good to also good to duplicate all the info in the thread.)

Reply:
sorry, but i'm not familiar with conda (assuming you're referring to https://conda.io/), so i really have no idea how to test it

the goal here isn't to make pkg-config required, but to make it so that if the package supports pkg-config, we only use that info

Reply:
Please put details in the commit messages!  Those are what help you in 5 years while checking git blame :wink: